tofhir-engine/src/main/scala/io/tofhir/engine/execution/RunningJobRegistry.scala

```diff
@@ -97,7 +97,10 @@ class RunningJobRegistry(spark: SparkSession) {
         updatedExecution.getStreamingQuery(mappingTaskName).awaitTermination()
       }catch{
         case exception: StreamingQueryException =>{
-          ExecutionLogger.logExecutionStatus(execution, FhirMappingJobResult.FAILURE, Some(mappingTaskName), Some(exception), isChunkResult = false)
+          val topicNames = execution.mappingTasks.find(mappingTask => mappingTask.name.contentEquals(mappingTaskName)).get.sourceBinding.map(source => source._2.asInstanceOf[KafkaSource].topicName).mkString(", ")
```

Comment:
Let's use getCause function to detect org.apache.kafka.common.errors.UnknownTopicOrPartitionException and then add the missing topics to the error message.

The first cause is ExecutionException and its cause is UnknownTopicOrPartitionException. You can add a match statement accordingly.